pkgs/development/python-modules/awkward-cpp/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, cmake
+, numpy
+, pybind11
+, scikit-build-core
+, typing-extensions
+}:
+
+buildPythonPackage rec {
+  pname = "awkward-cpp";
+  version = "2";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-XmP0PjE19224Hgkkp07PSHD1hcEan0MlaLN3wEAohow=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pybind11
+    scikit-build-core
+  ] ++ scikit-build-core.optional-dependencies.pyproject;
+
+  propagatedBuildInputs = [
+    numpy
+    typing-extensions
```

Comment:
According to [this line](https://github.com/scikit-hep/awkward/blob/main/awkward-cpp/pyproject.toml#L12), only `numpy` is needed.